modules/styling/lib/cs.ts

```diff
@@ -0,0 +1,355 @@
+ * ```
+ */
+export function createModifiers<T extends ModifierConfig>(input: T): ModifierReturn<T> {
+  const modifierFn = (modifiers: Partial<ModifierValues<T>>) => {
+    return Object.keys(modifiers)
+      .filter(key => input[key] && (input as any)[key][modifiers[key]])
+      .map(key => (input as any)[key][modifiers[key]])
+      .join(' ');
+  };
+
+  Object.keys(input).forEach(key => {
+    // @ts-ignore TypeScript makes it a pain to deal with index types
+    modifierFn[key] = input[key];
+  });
+
+  return modifierFn as ModifierReturn<T>;
+}
+
+function isCsPropsReturn(input: {}): input is CsToPropsReturn {
+  return input.hasOwnProperty('style') || input.hasOwnProperty('className');
+}
+
+/**
+ * All acceptable values of the `cs` prop. It can be a CSS class name, any CSS properties, an object
+ * with a `className` and `styles`, or an array of these
+ */
+export type CSToPropsInput =
+  | undefined
+  | CS
+  | CsToPropsReturn
+  | React.CSSProperties
+  | CSToPropsInput[];
+
+export type CsToPropsReturn = {className?: string; style?: React.CSSProperties};
+/**
+ * A function that takes in a single input, or an array. The type of the input is either:
+ *
+ * - `string` - it represents a CSS class name
+ * - `undefined` - there is no value. This is provided for convenience for developers to not have to
+ *   filter out undefined
+ * - `{--{hash}-{varName}: {value}}` - a `Map` of CSS variable to values
+ * - `{style: ..., className: ...}` an object already returned by another `csToProps` function call
+ *   (for nesting)
+ * - An array containing any of the above. This will recurse over each entry to produce a single,
+ *   reduced `{style: ..., className: ...}` object
+ * @param input
+ * @returns
+ */
+export function csToProps(input: CSToPropsInput): CsToPropsReturn {
+  // input is a string, so it must only be a class name
+  if (typeof input === 'string') {
+    return {className: input};
+  }
+
+  // input isn't defined, so we'll return an empty object
+  if (!input) {
+    return {};
+  }
+
+  // A nested `csToProps`, we'll simply return with no modification
+  if (isCsPropsReturn(input)) {
+    return input;
+  }
+
+  // At this point, `input` is either an array or a `Record<T, string>`. If it isn't an array, it
+  // must be a style object for setting CSS variables. So set the `style` attribute
+  if (!Array.isArray(input)) {
+    return {style: input};
+  }
+
+  // An array. it is the only thing left. We iterate and recurse over the array to produce a single
+  // object with `style` and `className` attributes to spread on an element
+  return input.map(csToProps).reduce((result, val) => {
+    return {
+      className: [result.className, val.className].filter(v => v).join(' '),
+      style: {...result.style, ...val.style},
+    };
+  }, {} as CsToPropsReturn);
+}
+
+export interface CSProps {
+  /** The `cs` prop takes in a single value or an array of values. You can pass the CSS class name
```

Comment:
```suggestion
  /** The `cs` prop takes in a single value or an array of values. You can pass the CSS class name
```
Just through out the file, are we using `cs` or `createStyles`?